### PR TITLE
Docker

### LINKED ISSUE #69: Deployment flow
This probably means dockerization??

### PR DESCRIPTION
**Pull request by mouse-reeve**
Fixes #69
Works on #91 and #40

### CODE REVIEW

**Comment by mouse-reeve** [comment]
I'm getting the error:
`ERROR: No matching distribution found for celery==4.4.2 (from -r requirements.txt (line 1))`
when I try to deploy on ubuntu so I would not yet consider it a repeatable deploy.

**Comment by mouse-reeve** [comment]
@cthulahoops I'm still figuring out how to make this run consistently and how to actually use celery, but it's mostly working. since it'll effect everyone's workflow, I welcome however much review/input you have and are interested in giving, but of course no pressure!

**Comment by cthulahoops** [comment]
This looks reasonable to me. I haven't managed to get it build yet. I'll have another look in the morning and see if I can get it working.

In general, I'd prefer to have docker as an option rather than the only way to run the project as I've found it a bit unwieldy in development. I'm not sure how practical that is once we're depending on celery, rabbitmq, etc. I'm not very familiar with celery either.

**Comment by mouse-reeve** [comment]
Agreed on having docker be optional. I don't know for sure yet if I can keep celery as an optional/prod-only dependency

**Comment by cthulahoops** [comment]
> Agreed on having docker be optional. I don't know for sure yet if I can keep celery as an optional/prod-only dependency

Oh yeah! I hadn't thought of that, but it's got to be possible

Similarly for local messages activitypub allows us to bypass https and deliver the messages internally which might make testing easier.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: '3'
+
+services:
+    db:
+        image: postgres
+        env_file: .env
+        volumes:
+          - pgdata:/var/lib/posgresql/data
```
Comment (cthulahoops) [inline docker-compose.yml]: Should be: ```/var/lib/pos**t**gresql/data```?

Even after fixing I get stuck at:

```
db_1             | running bootstrap script ... ok
db_1             | performing post-bootstrap initialization ... 2020-03-24 14:28:54.434 UTC [40] FATAL:  zero-length delimited identifier at or near """" at character 12
db_1             | 2020-03-24 14:28:54.434 UTC [40] STATEMENT:  ALTER USER ""postgres"" WITH PASSWORD E'"fedireads"';
db_1             | 	
db_1             | child process exited with exit code 1
db_1             | initdb: removing contents of data directory "/var/lib/postgresql/data"
fedireads_db_1 exited with code 1
```

Comment (mouse-reeve) [reply docker-compose.yml]: after my last commit that fixes this, I re-ran `docker-compose build` and `docker-compose up` and this error went away for me🤞

Comment (mouse-reeve) [reply docker-compose.yml]: I also changed my database type in .env from sqlite to postgres, which may be related

Comment (cthulahoops) [reply docker-compose.yml]: Yeah, it's working for me too now. :)

File: README.md [hunk 2]
```diff
@@ -45,19 +45,42 @@ This project is still in its very early stages, but these are the higher-level f
...
 
+### Docker
+You can also run the application in a docker container. You'll have to install the Docker and docker-compose:
+
+```bash
+docker-compose build
+docker-compose up
+docker-compose exec web python manage.py migrate
+```
+
```
Comment (cthulahoops) [inline README.md]: Need instructions to invoke init_db too? (Not sure the best way: ```docker-compose exec web python manage.py shell -c 'import init_db'``` perhaps?)

**Pull request merged**